test/functional/lua/json_spec.lua

```diff
@@ -152,6 +152,23 @@ describe('vim.json.encode()', function()
     clear()
   end)
 
+  it('dumps strings with escaped slash', function()
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/')]]))
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/', {})]]))
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/', { __invalid_key = true })]]))
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/', { escape_forward_slash = true })]]))
+  end)
+
+  it('dumps strings without escaped slash', function()
+    eq('"Test/"', exec_lua([[return vim.json.encode('Test/', { escape_forward_slash = false })]]))
+    eq(
+      '"Test/"',
+      exec_lua(
+        [[return vim.json.encode('Test/', { __invalid_key = true, escape_forward_slash = false })]]
```

Comment:
all these tests should be in 1 test, especially to check that the option does not affect anything globally (and there should be an explicit check for that)

Reply:
Merged into 1 test